app/scripts/controllers/swaps.types.ts

```diff
@@ -121,6 +323,45 @@ export type QuoteRequest = {
   walletAddress: string;
 };
 
+export type SwapsControllerOptions = {
+  getBufferedGasLimit: (
+    params: {
+      txParams: {
+        value: string;
+        data: string;
+        to: string;
+        from: string;
+      };
+    },
+    factor: number,
+  ) => Promise<{ gasLimit: string; simulationFails: boolean }>;
+  provider: ExternalProvider | JsonRpcFetchFunc;
+  getProviderConfig: () => ProviderConfig;
```

Comment:
Can this be removed? It doesn't seem like `providerConfig` is directly referenced anywhere in the controller after being assigned to a class field.

`providerConfig` is also removed in the next network-controller release (v20).

Reply:
You are right that its not called internally, but i kept it because it gets exposed as a public method. Did some research and it seems that this is not getting called from anywhere in the codebase so seems safe to remove